tests/example_data/case_dependencies.py

```diff
@@ -0,0 +1,147 @@
+"""
+Test ability to ensure that tests can rely on other tests being executed first
+and in the proper order
+
+Output is tracked and an exception will be thrown if it does not match.
+Run with the following command:
+
+spektrum -s tests -p case_dependencies --select-tests "re:third.*","re:fourth.*"
+"""
+
+from spektrum import Spec, DataSpec
+from spektrum import depends_on, fixture
+
+
+@fixture
+class DependenciesFixture(Spec):
+    async def before_all(self):
+        await super().before_all()
+        self.output = []
+
+    async def after_all(self):
+        expected_output = [
+            'first',
+            'second',
+            'third',
+            'fourth',
+        ]
+        if self.output != expected_output:
+            raise Exception('output does not match')
```

Comment:
The `after_all` exception thrown inhibits the ability to test different scenarios such as `re:third` to ensure only `first`,`second`, and `third` tests get run.  I feel the `after_all` sections should be removed to allow for testing beyond just running the whole thing or just `fourth`

Reply:
So, what I was trying to do, was slightly change each Spec so it covered all the possible scenarios of how `depends_on()` could be used, so you wouldn't have to run multiple combinations via the command line... and then the `expected_output` and exception would do all the work for you, so you didn't have to manually compare what happened vs what should have happened.

If that doesn't seem intuitive though, I can change this to work the same way the `select_tests.py` file works, where you have to try different combinations via the command line and manually check to see if it's executing as expected.